src/server.c

```diff
@@ -2746,8 +2746,8 @@ void initServer(void) {
         listSetFreeMethod(server.db[j].defrag_later,(void (*)(void*))sdsfree);
     }
     server.rehashing = listCreate();
-    server.db_hashtable_overhead_total = 0;
-    server.db_hashtable_overhead_rehashing = 0;
+    server.overhead_hashtable_lut = 0;
+    server.overhead_hashtable_rehashing = 0;
```

Comment:
don't we also need to zero these when a dict is deleted or detached?
i.e. `emptyDbStructure()`, which will take care of FLUSHDB SYNC/ASYNC, and REPL_DISKLESS_LOAD_SWAPDB

or maybe it's a better idea to just call the `rehashingCompleted` callback from `_dictClear()`?
i.e. we're already have some leaks in the bucket counters?
@hpatro PTAL.

Reply:
> don't we also need to zero these when a dict is deleted or detached?

Yes we need, and I do want to zero them in `_dictClear()`. It just took me some time to make sure all cases are covered.

Reply:
Or, we may have a new callback in dictType that is called when dict is cleared/released?

Reply:
maybe we can just call the existing callbacks.
i.e. if we're in the middle of rehashing, trigger `rehashingCompleted`.
and after that, call rehashingStarted and rehashingCompleted with destination size of 0?

Reply:
I edit the metrics in `emptyDbStructure()` bacause `_dictClear()` may involve too many other structures.

And I prefer to modify the metric derectly instead of calling `rehashingStarted` and `rehashingCompleted`, because most of their job is already done in `emptyDbStructure()`.

Reply:
i see we already handle `server.rehashing` and `bucket_count` in emptyDbStructure, so i guess we're good to go.
in theory there's some room for messing up things when we deal with server struct globals, because in REPL_DISKLESS_LOAD_SWAPDB there are several dictionaries being handled, but it look like the code handles them correctly (by removing what's in the dict being released, rather than zeroing variables)